students/minchan/westagram/users/views.py

```diff
@@ -1,3 +1,93 @@
-from django.shortcuts import render
+import json
+import re
+from django.http.response   import JsonResponse
+from django.views           import View
+from users.models           import Users
+from django.http            import HttpResponse
 
 # Create your views here.
+class user_register(View) :
+    def valid_email(self,email) :
+        email_regular       = re.compile("[@][a-zA-Z]*[.]")
+        if email_regular.search(email) == None :
+            return False
+        return True
+
+
+    def is_email_duplicate(self,email) :
+        if Users.objects.filter(email=email) :
+            return True # 중복값이 있으면 True를 반환.
+        return False
+    
+
+    def valid_pw(self,pw) :
+        # 검사항목 - 0: 8자리 이상, 1:문자포함, 2:숫자포함, #:특수문자포함
+        test_case           = [0,0,0,0]
+        alphabet_regular    = re.compile("[a-zA-Z]")
+        number_regular      = re.compile("[\d]")
+        symbol_regular      = re.compile("[`~!@#$%^&*(),<.>/?]") #더 많은 특수문자의 허용은 [^0-9a-zA-Z]로 대체 가능.
```

Comment:
1. Class 명은 CamelCase로 만들어 주세요.
2. . complie은 member 변수로 선언해서 사용해 주세요.
3. 함수명은 `동사`로 만들어 주세요.
4. `if Users.objects.filter(email=email):` 에서는 `.exists()` method를 chaining해서 사용해서 boolean만 반환할 수 있도록 만들어 주세요.
5. 숫자포함, 특수문자 포함, 문자포함을 한번에 처리할 수 있는 정규식을 사용해주세요. for문이 너무 많이 사용되고  있습니다.

```suggestion
class User(View) :
    REGEX_EMAIL = re.compile("[@][a-zA-Z]*[.]")
    REGEX_PASSWORD = re.compile(... 숫자포함, 특수문자 포함, 문자포함을 한번에 처리할 수 있는 정규식을 사용해주세요.)
    
    def validate_email(self,email) :    
        if not REGEX_EMAIL.match(email):
            return False
            
        return True

    def is_email_duplicated(self, email) :
        if Users.objects.filter(email=email).exists():
            return True 
            
        return False
    
    def validate_password(self, password) :
```

Reply:
클래스명 변경, 함수명 변경, 
compile 정규식 한문장으로 상수로 선언해서 사용했습니다.
이메일 검사 시 exists()처리 했습니다.

return 시에 True, False를 반환하는 exist()는 그대로 반환하고
다른 검사용 함수는 return True if ~~ else Fasle 로, 한줄로 처리했습니다.

Reply:
감사합니다!